src/Microsoft.OData.Core/ODataNotificationReader.cs

```diff
@@ -113,18 +121,44 @@ public override Task<string> ReadToEndAsync()
         /// <param name="disposing">True if called from Dispose; false if called from the finalizer.</param>
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            if (!this.disposed && disposing)

```

Comment:
Thank you for adding this! I'm actually going to post next week in the maintainability channel more about disposing. However, this does *now* follow the MSDN guidelines for implementing dispose, I love it!